plugin.video.montreal.greek-tv/addon.py

```diff
@@ -0,0 +1,283 @@
+
+    if menu is None:
+        return
+
+    directory.add(menu, content='images')
+
+
+def _paper_index(link):
+
+    base_img_url = 'https://image.isu.pub/'
+
+    html = requests.get(link).text
+
+    script = parseDOM(html, 'script', attrs={'type': 'application/javascript'})[-2]
+
+    data = json.loads(script.partition(' = ')[2].rstrip(';'))
+    document = data['document']
+    total_pages = int(document['pageCount'])
+
+    menu = []
+
+    for page in list(range(1, total_pages + 1)):
+
+        title = document['title'] + ' - ' + control.lang(30003) + ' ' + str(page)
+        page_img = base_img_url + document['id'] + '/jpg/page_{0}_thumb_large.jpg'.format(str(page))
+        page_url = base_img_url + document['id'] + '/jpg/page_{0}.jpg'.format(str(page))
+
+        data = {'title': title, 'image': page_img, 'url': page_url}
+
+        menu.append(data)
+
+    return menu
+
+
+def paper_index(link):
+
+    menu = []
+
+    items = cache.get(_paper_index, 12, link)
+
+    if items is None:
+        return
+
+    for i in items:
+        li = control.item(label=i['title'])
+        li.setArt(
+            {
+                'poster': i['image'], 'thumb': i['image'],
+                'fanart': control.join(control.addonPath, 'resources', 'media', 'newspaper_fanart.png')
+            }
+        )
+        li.setInfo('image', {'title': i['title'], 'picturepath': i['url']})
+        url = i['url']
+        menu.append((url, li, False))
+
+    control.content(syshandle, 'images')
+    control.addItems(syshandle, menu)
+    control.directory(syshandle)
+
+
+def _podcasts():
+
+    menu = []
+
+    feed_url = 'http://greektimes.ca/feed/podcast/'
+
+    xml = requests.get(feed_url).text
+
+    items = parseDOM(xml, 'item')
+
+    for item in items:
+
+        title = parseDOM(item, 'title')[0]
+        uri = parseDOM(item, 'enclosure', attrs={'type': 'audio/mpeg'}, ret='url')[0]
+        fanart = parseDOM(item, 'enclosure', attrs={'type': 'image/(?:jpeg|png)'}, ret='url')[0]
+        image = parseDOM(item, 'img', attrs={'class': '.*?wp-image-\d{1,4}.*?'}, ret='srcset')[0]
+        img_urls = image.split(',')
+        image = [
+            i.rpartition(' ')[0].strip() for i in img_urls if int(i[-5:-1]) == min([int(v[-5:-1]) for v in img_urls])
+        ][0]
+        comment = parseDOM(item, 'description')[0]
+        year = int(re.search('(\d{4})', parseDOM(item, 'pubDate')[0]).group(1))
+
+        data = {
+            'title': title, 'url': uri, 'image': image, 'fanart': fanart, 'comment': comment, 'lyrics': comment,
+            'year': year
+        }
+
+        menu.append(data)
+
+    return menu
+
+
+def podcasts():
+
+    items = cache.get(_podcasts, 6)
+
+    if items is None:
+        return
+
+    for i in items:
+        i.update({'action': 'play', 'isFolder': 'False'})
+
+    directory.add(items, content='music')
+
+
+def main_menu():
+
+    xml = requests.get(url='http://s135598769.onlinehome.us/mgtv.xml').text
```

Comment:
This is not allowed. You need to parse the webpage to find the stream url, or at least to get the reference to this url so that we know it actually comes from greektimes website.

Reply:
This url was provided by montreal greek tv administrator. He wants to provide stream urls and names via this xml.

Reply:
I confirm, this URL was provided by me (owner/administrator of The Montreal Greek Times) for the TV feed.

Please see [private] statement I made on the GreekTimes website that confirms that the source is from the GreekTimes web page.

here:
http://greektimes.ca/1189/

 Thank you,

Dimitri Padopoulos, owner
The Montreal Greek Times